Civi/API/Subscriber/DynamicFKAuthorization.php

```diff
@@ -367,10 +372,34 @@ public function getCustomFields() {
         'field_name' => $query->field_name,
         'table_name' => $query->table_name,
         'extends' => $query->extends,
+        'column_name' => $query->column_name,
         'entity_table' => CoreUtil::getTableName($query->extends),
       ];
     }
     return $rows;
   }
 
+  /**
+   * @param int $id
+   *   e.g. file ID.
+   * @return array
+   */
+  private function getCustomFieldReferencesToId($id): array {
+    $rows = [];
+    foreach ($this->getCustomFields() as $field) {
```

Comment:
For systems with... say... 0-10 custom-file-fields... that's an easy loop?

And for systems with... say... 100+ custom-file-fields... eh... :grimacing: :hear_no_evil: 

Anyway, this makes sense. It's what you need for the 6.3 schema.